indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -337,6 +337,28 @@ public Response apply(TaskQueue taskQueue)
     );
   }
 
+  @POST
+  @Path("/task/{dataSource}/shutdownAllTasks")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response shutdownTasksForDataSource(@PathParam("dataSource") final String dataSource)
+  {
+    return asLeaderWith(
+        taskMaster.getTaskQueue(),
+        new Function<TaskQueue, Response>()
+        {
+          @Override
+          public Response apply(TaskQueue taskQueue)
+          {
+            final List<TaskInfo<Task, TaskStatus>> tasks = taskStorageQueryAdapter.getActiveTaskInfo(dataSource);
+            for (final TaskInfo<Task, TaskStatus> task : tasks) {
```

Comment:
Maybe `tasks.parallelStream().forEach(t -> taskQueue.shutdown(t.getId()))` will be better.

Reply:
I do not think so, `taskQueue.shutdown` is a synchronized operation. Use `parallelStream ` has no performance improvement.

Reply:
Yes, you are right. Even if `ReentrantLock` is good at handling with concurrent scenes, we should not try to use `parallelStream` here. I didn't notice it. My fault. Forget it, what do you think of converting it into a more functional programming style, such as `tasks.forEach(t -> taskQueue.shutdown(t.getId()))`?

Reply:
I think there is no need. Other `for loop` code in class `OverlordResource` are not functional programming style. Coding style should keep consistency.

Reply:
Okay, it's up to you. :sweat_smile:

Reply:
Agree that it's good to be consistent within a file, and the rest of this file is totally down with for loops.